linear_algebra/src/rank_of_matrix.py

```diff
@@ -0,0 +1,89 @@
+"""
+Calculate the rank of a matrix.
+
+See: https://en.wikipedia.org/wiki/Rank_(linear_algebra)
+"""
+
+
+def rank_of_matrix(matrix: list[list[int]]) -> int:
```

Comment:
```suggestion
def rank_of_matrix(matrix: list[list[int | float]]) -> int:
```
This is required due to your multiplication returning a float